src/input/scroll.c

```diff
@@ -113,6 +97,190 @@ int scroll_get_direction(const mouse *m)
     return DIR_8_NONE;
 }
 
+static void clear_scroll_decay(const map_point position)
+{
+    data.speed.x = 0;
+    data.speed.y = 0;
+    data.speed.last_time = time_get_millis();
+    data.speed.last_position = position;
+    data.speed.decaying = 0;
+}
+
+static void find_scroll_speed(map_point position)
+{
+    int current_time = time_get_millis();
+    int time_delta = current_time - data.speed.last_time;
+
+    int position_delta;
```

Comment:
Variable definition can be moved to assignment, below the if statement (we're not C89 anymore :D )